csfieldguide/static/interactives/pixel-viewer/js/pixel-viewer.js

```diff
@@ -996,14 +1019,34 @@ var paint = function(row, col, left, top, width, height, zoom) {
             context.fillStyle = "rgba(110, 110, 110, " + text_opacity + ")";
         }
 
-        // Pretty primitive text positioning :)
-        var cell_lines = cell_text.split('\n');
-
-        for (var i = 0; i < cell_lines.length; i++)
-            context.fillText(cell_lines[i] + pixelData[i], left + (6 * zoom), top + (14 * zoom) + (i * cell_line_height * zoom) );
+        if (show_hex) {
+          context.font = (10 * zoom).toFixed(2) + 'px Consolas, Courier New, monospace';
+          r = pixelData[0];
+          g = pixelData[1];
+          b = pixelData[2];
+          hex_string = rgbToHex(r, g, b);
+          context.fillText(hex_string, left + (4 * zoom), top + (14 * zoom) + (cell_line_height * zoom));
+        } else {
+          context.font = (14 * zoom).toFixed(2) + 'px Consolas, Courier New, monospace';
+          // Pretty primitive text positioning
+          var cell_lines = cell_text.split('\n');
+          for (var i = 0; i < cell_lines.length; i++) {
+            context.fillText(cell_lines[i] + pixelData[i], left + (6 * zoom), top + (14 * zoom) + (i * cell_line_height * zoom));
+          }
+        }
     }
 };
 
+// Taken from https://stackoverflow.com/questions/5623838/rgb-to-hex-and-hex-to-rgb
+function componentToHex(c) {
+  var hex = c.toString(16);
+  return hex.length == 1 ? "0" + hex : hex;
+}
+
+function rgbToHex(r, g, b) {
+  return "#" + componentToHex(r) + componentToHex(g) + componentToHex(b);
```

Comment:
Looks like Tim was wanting individual R G and B components like before
> The default in both cases should still be decimal, but they need an option to show the numbers 0..255 as 00..FF. At this stage I don't think they should have '0x' in front of them.

So I guess that'd mean an option between decimal and hexadecimal - it makes more sense too